core/src/com/unciv/ui/worldscreen/WorldScreen.kt

```diff
@@ -208,12 +209,31 @@ class WorldScreen(val viewingCiv:CivilizationInfo) : CameraStageBaseScreen() {
                 viewingCiv.tradeRequests.isNotEmpty() -> TradePopup(this)
             }
         }
+
+         if  (canAutoCompleteTurn()) {
+            nextTurn()
+         }
+
         updateNextTurnButton(hasVisibleDialogs()) // This must be before the notifications update, since its position is based on it
         notificationsScroll.update(viewingCiv.notifications)
         notificationsScroll.setPosition(stage.width - notificationsScroll.width - 5f,
                 nextTurnButton.y - notificationsScroll.height - 5f)
     }
 
+    private fun canAutoCompleteTurn(): Boolean {
+        return  !hasVisibleDialogs() &&
+                isPlayersTurn &&
+                !waitingForAutosave &&
+                UncivGame.Current.settings.autoCompleteTurn &&
+                !viewingCiv.policies.canAdoptPolicy() &&
+                viewingCiv.greatPeople.freeGreatPeople == 0 &&
+                viewingCiv.popupAlerts.none() &&
+                viewingCiv.tradeRequests.isEmpty() &&
+                !viewingCiv.shouldOpenTechPicker() &&
+                viewingCiv.getIdleUnits().isEmpty() &&
```

Comment:
Isn't there a "IsSomethingOpen" function that covers many of these? You could probably use that

Reply:
I will have to go back and give this a look another time, I definitely want the readability of this function to be better.